Fix tooltip label name rendering in pie charts.

lukeelmers opened the pull request:
Resolves #39788
Resolves #39655 

### Summary

In the process of moving visualizations to be powered by the interpreter, there was a large refactoring to field formatting in #28746, in which the `collectBranch` function -- which is used to gather the data necessary to render names in tooltips -- was accidentally missed.

As a result, labels in all pie charts have been displaying the fallback text since 7.2. This addresses the issue by implementing the fixes in `collectBranch` that were meant to be included in #28746. 

I've also added some unit tests here to cover this case, and converted the existing ones to jest.

Lastly, I co-located the function with the hierarchical tooltip formatter, which is the only place where it is actually used.

### Reviewers

Since I moved some files around as a part of this, it may be easier to review commit-by-commit.

### QA

To verify this works as expected, create a pie chart with multiple buckets, however over the slices, and confirm that the expected label names appear in the tooltip instead of labels like `level 1`, `level 2`, etc

elasticmachine commented:
Pinging @elastic/kibana-app-arch

lukeelmers commented on src/legacy/ui/public/agg_response/hierarchical/build_hierarchical_data.js:
```diff
@@ -30,7 +30,6 @@ export const buildHierarchicalData = (table, { metric, buckets = [] }) => {
     slices = [{
       name: metricColumn.name,
       size: table.rows[0][metricColumn.id],
-      aggConfig: metricColumn.aggConfig
```
Since these are always `undefined` I went ahead and removed them here so we aren't passing `aggConfig: undefined` around everywhere.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5172/)

lukeelmers commented:
@wylieconlon So the root cause was essentially that, in #28746 when we refactored the field formatters, we modified the hierarchical response handler so that it doesn't require `aggConfig`, and instead passes the formatted label as the `name` within the tabified column data.

However, since the `collectBranch` function simply used a fallback when `aggConfig` was unavailable (and since there weren't tests for each of these different cases), the labels silently failed, reverting to the fallback instead.

So basically this was an area we overlooked in that refactor, which was intended to pull out all usages of `aggConfig` from this part of the code.

Aside from that, I'm not super familiar with the original implementation of this function, so I have some questions too (for example, I'm not sure if we should _ever_ be reaching the case where the fallback text is used... maybe the function really should be throwing if we ever get there).

lukeelmers commented on src/legacy/ui/public/vis/components/tooltip/_collect_branch.js:
```diff
@@ -17,26 +17,27 @@
+        const { column, table } = i.rawData;
+        return table.columns[column].name;
+      }
+      return i.name || `level ${i.depth}`;
```
`i.name` is needed here so we can get the `"Count"` label back for metrics, but TBH I'm not sure when/if we should ever expect to get the fallback `"level n"` text. AFAICT that code should be unreachable.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5465/)

ppisljar commented:
retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5547/)
